language/stdlib/modules/transaction_fee.move

```diff
@@ -1,105 +1,87 @@
+        Transaction::assert(Signer::address_of(libra_account_singleton_addr) == 0xA550C18, 0);
+        FeePaymentCapability {}
+    }
 
-        while (index < num_validators) {
+    public fun pay_fee<CoinType>(fee: Libra<CoinType>, _cap: &FeePaymentCapability)
+    acquires CollectedFees, LBRIdent {
+        // For LBR we need to extract the underlying coins from the
+        // on-chain reserve and then push these to the
+        if (exists<LBRIdent<CoinType>>(0xFEE)) {
 
-            let addr = LibraSystem::get_ith_validator_address(index);
-            // Increment the index into the validator set.
-            index = index + 1;
+            // The cointype is LBR, but we need to cast CoinType to LBR in
+            // order to extract the underlying coins. We do this by going
+            // through storage.
+            let lbr_ident = borrow_global_mut<LBRIdent<CoinType>>(0xFEE);
+            let value = Libra::value(&fee);
+            Libra::deposit(&mut lbr_ident.coin, fee);
+            let lbr_fee = Libra::withdraw<LBR>(
+                &mut borrow_global_mut<LBRIdent<LBR>>(0xFEE).coin,
+                value,
+            );
 
-            LibraAccount::pay_from_capability<Token>(
-                addr,
-                &distribution_resource.fee_withdrawal_capability,
-                amount_to_distribute_per_validator,
-                x"",
-                x""
+            let (coin1, coin2) = LBR::unpack(lbr_fee);
+            let collected_fee_coin1 = borrow_global_mut<CollectedFees<Coin1>>(0xFEE);
+            Libra::preburn_with_resource(
+                coin1,
+                &mut collected_fee_coin1.preburn,
+                0xFEE
             );
-           }
+            let collected_fee_coin2 = borrow_global_mut<CollectedFees<Coin2>>(0xFEE);
+            Libra::preburn_with_resource(
+                coin2,
+                &mut collected_fee_coin2.preburn,
+                0xFEE
+            );
+        } else {
+            let collected_fee = borrow_global_mut<CollectedFees<CoinType>>(0xFEE);
+            Libra::preburn_with_resource(
+                fee,
+                &mut collected_fee.preburn,
+                0xFEE
+            )
+        }
     }
 
-    // This calculates the amount to be distributed to each validator equally. We do this by calculating
-    // the integer division of the transaction fees collected by the number of validators. In
-    // particular, this means that if the number of validators does not evenly divide the
-    // transaction fees collected, then there will be a remainder that is left in the transaction
-    // fees pot to be distributed later.
-    fun per_validator_distribution_amount(amount_collected: u64, num_validators: u64): u64 {
-        Transaction::assert(num_validators != 0, 0);
-        let validator_payout = amount_collected / num_validators;
-        validator_payout
+    public fun burn_fees<CoinType>(burn_cap: &Libra::BurnCapability<CoinType>)
```

Comment:
Hmm, interesting requirements question here. This lets us burn fees in only one currency at a time; I'm not sure if what we want is that or "burn fees in all currencies at once" (we should ask). The latter might actually be easier to implement; it will let us get rid of all the special casing for LBR.